politeiawww/cmd/politeiavoter/config.go

```diff
@@ -47,12 +48,14 @@ const (
 )
 
 var (
-	defaultHomeDir        = dcrutil.AppDataDir("politeiavoter", false)
-	defaultConfigFile     = filepath.Join(defaultHomeDir, defaultConfigFilename)
-	defaultLogDir         = filepath.Join(defaultHomeDir, defaultLogDirname)
-	defaultVoteDir        = filepath.Join(defaultHomeDir, defaultVoteDirname)
-	dcrwalletHomeDir      = dcrutil.AppDataDir("dcrwallet", false)
-	defaultWalletCertFile = filepath.Join(dcrwalletHomeDir, "rpc.cert")
+	defaultHomeDir    = dcrutil.AppDataDir("politeiavoter", false)
+	defaultConfigFile = filepath.Join(defaultHomeDir, defaultConfigFilename)
+	defaultLogDir     = filepath.Join(defaultHomeDir, defaultLogDirname)
+	defaultVoteDir    = filepath.Join(defaultHomeDir, defaultVoteDirname)
+	dcrwalletHomeDir  = dcrutil.AppDataDir("dcrwallet", false)
+	defaultWalletCert = filepath.Join(dcrwalletHomeDir, walletCertFile)
```

Comment:
How about using our own config dir here (`politeiavoter`) and name the cert file `dcrwallet.cert`?

I would follow the dcrwallet approach here and not dcrctl. dcrwallet has a copy of the external cert in [its own config dir](https://github.com/decred/dcrwallet/blob/b00d7944519ad9da744d73e73cc18d6e9b74558f/sample-dcrwallet.conf#L54), which becomes fully independent of the other software.

dcrctl, in contrast, is borrowing an [external cert](https://github.com/decred/dcrctl/blob/da63da7aa7bf71e1fd9fdb98dd83b3857e4a5f92/sample-dcrctl.conf#L35).

The practical difference is I don't need to touch the `cafile` param in dcrwallet and control everything with a single `--appdata` switch (good), but for dcrctl I have to set `rpccert` to an absolute path (one more thing to configure + not pretty).

Reply:
To be fair, I understand that just reading other apps' config files feels like _less_ initial copying of certs to get up and running, but somehow it did not work for me and dcrwallet's autonomus approach feels more robust.

We can get the best of both worlds too, at the cost of complexity, by adding one more place to the lookup algo:

1. if `--walletgrpccert` specified, use that
2. else use `walletgrpccert` from effective config
3. else use `dcrwallet.cert` from effective AppData dir
4. else use `~/.dcrwallet/rpc.cert`

Reply:
Combining the approaches, while it does cover all the basis, is overly complex.  Using the external cert works out of the box when using the default dcrwallet application directory.

If you're using custom application directories then you should expect to have to adjust config settings.

>  but somehow it did not work for me

Why not?  It should.  If it doesn't then it points to something else being wrong.

Reply:
Yes I'm using a custom dcrwallet application directory.

It "did not work" in a sense it was confusing to set up. Of all CLI binaries dcrwallet was the easiest to use with custom dirs since everything can be flipped with a single `--appdata` switch (which is great when managing multiple wallets). After that UX I felt using `dcrwallet.cert` over `~/.dcrwallet/rpc.cert` would be an improvement (for the custom dir use case).

I can live with setting it via `walletgrpccert` too, I just didn't like absolute paths in it. IIRC relative path like `walletgrpccert=rpc.cert` is resolved relative to program's current directory and not HomeDir (`politeiavoter`).

Reply:
I'll get marco's opinion on this.  The dcrwallet method vs the dcrctl method.

Reply:
Curious what you guys decided re the dcrwallet method?